### DISCUSSION TITLE
No permission to write to the Hosts file

### DISCUSSION THREAD

**Discussion by hnevarez89** [oldj/SwitchHosts#898] [Q&A]
Hi, I have a problem with SwitchHosts.
I added a remote host URL to update automatically. The problem is that it is not doing so, when enabling the added host I get the following message: No permission to write to the Hosts file.
![2024-09-12_083923](https://github.com/user-attachments/assets/2a12f84f-0d83-4cae-a067-6324cc8e3e55)

**Comment by ghost** [comment]
quit the app and run as administrator

**Comment by xyxzxyz** [comment]
in addition to @Potato-007's answer, make sure hosts file isn't 'read only'. For whatever reason, my hosts file was set to 'read only', so even running as administrator, it still prompts no permission error.

**Reply by sam619007** [reply]
@xyxzxyz How to disable 'read only' mode?

**Reply by xyxzxyz** [reply]
Go to properties of the hosts file and uncheck read only box in attributes section.

**Comment by a1ch3mist** [comment]
Same problem here. Ran as admin, and also verified file is not in read only mode.

What else can I do?

**Reply by Mystard** [reply]
Check that your antivirus isn't blocking the file change. For example Kaspersky blocks hosts file edits by default from apps that are not in the trusted group. The default is also block instead of ask the user, so there is no popup or anything. You can see them being blocked in reports / logs though, under Intrusion Prevention.

Giving SwitchHosts the permission to edit the hosts file should fix it. For example in Kaspersky, this is done under Intrusion Prevention -> Manage Applications -> SwitchHosts -> Files and system registry -> Critical settings -> hosts. Here you can set the write access to allow.
